features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/Components/Commons.jsx

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { Component } from 'react';
+import 'react-toastify/dist/ReactToastify.min.css';
+import Button from '@material-ui/core/Button';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import Grid from '@material-ui/core/Grid';
+import { withStyles } from '@material-ui/core/styles';
```

Comment:
Instead of calling  "import { withStyles } from '@material-ui/core/styles'; "
we can use as "import withStyles from '@material-ui/core/styles/withStyles';"

Reply:
@prasa7 , I think named import is safe here, since it won't make whole styles to be imported because of the [tree shaking](https://webpack.js.org/guides/tree-shaking/) effect, But there is an know issue(s) https://github.com/mui-org/material-ui/issues/10857 https://github.com/mui-org/material-ui/issues/5533  with `@material-ui/icons/` library , where tree shaking is now working properly, So that they have suggested not to use named import **only** in the `@material-ui/icons` library imports please reffer [MUI docs](https://material-ui.com/style/icons/#__next) for details

Reply:
@tmkasun: Thanks for the in depth explanation. Will keep this as it is.